src/init/bitcoin-node.cpp

```diff
@@ -26,6 +39,15 @@ class BitcoinNodeInit : public interfaces::Init
     {
         m_node.args = &gArgs;
         m_node.init = this;
+        // Extra initialization code that runs when a bitcoin-node process is
+        // spawned by a bitcoin-gui process, after the ArgsManager configuration
+        // is transferred from the parent process to the child process.
+        m_ipc->context().init_process = [this] {
```

Comment:
In commit "Make bitcoin-gui spawn a bitcoin-node process" (2516d397d54b021e37d6e7628655fe7c54d2fb06):

re: https://github.com/bitcoin/bitcoin/commit/2516d397d54b021e37d6e7628655fe7c54d2fb06#r53368428

> IIUC, it's going to be call at the first invocation on the server from the client as it's dependent on the whole `serverInvoke()`/`MakeServerField()`/etc call chain ? Sometimes it's hard to follow which objects are defined in `bitcoin`, which ones are in `libmultiprocess` and which ones are dynamically generated by `mpgen`. Though i agree it's fine for now, we can improve with time.

`init_process` is bitcoin-specific and libmultiprocess can't access it or reference it, and it isn't related to IPC method invocation (`serverInvoke` is a libmultiprocess internal function that bitcoin code shouldn't access and doesn't reference which translates an incoming IPC method call from a remote process to a C++ method call on a local object.)

`init_process` is a hack to deal with the fact that bitcoin code uses a bunch of global variables that depend on the `gArgs` global variable and they need to be initialized at some point.

When bitcoin is running with wallet & node & gui code in the same process, global variables only have to be initialized once by existing init code. But if the node process spawns a wallet process, or if the GUI process spawns a node process, then there are multiple copies of the global variables in different processes, and they have to be newly initialized in each spawned process, and existing init code is no longer sufficient. That is why the `GlobalArgs` class and `init_process` callback are introduced. `GlobalArgs` transfers `gArgs` state from the parent process to the child process and `init_process` sets up logging and other global variables that depend on `gArgs`.

Ideally there would be no global variables shared between node wallet and GUI code (they would move into `NodeContext` and `WalletContext` objects) and the `init_process` hack could go away. Short of that, I maybe rename `init_process` or you could think of it as something like `init_bitcoin_shared_global_variables_in_newly_spawned_child_process_after_transferring_gargs`